File: app/src/main/java/com/example/clever/view/home/cal/SelectDecorator.java
package com.example.clever.view.home.cal;

import android.app.Activity;
import android.graphics.drawable.Drawable;

import com.example.clever.R;
import com.prolificinteractive.materialcalendarview.CalendarDay;
import com.prolificinteractive.materialcalendarview.DayViewDecorator;
import com.prolificinteractive.materialcalendarview.DayViewFacade;

public class SelectDecorator implements DayViewDecorator {

    private final Drawable drawable;

    public SelectDecorator(Activity context) {
        this.drawable = context.getResources().getDrawable(R.drawable.calendar_select);
    }

    @Override
    public boolean shouldDecorate(CalendarDay day) {
        return true;
    }

    @Override
    public void decorate(DayViewFacade view) {
        view.setSelectionDrawable(drawable);
    }
}

File: app/src/main/java/com/example/clever/view/home/cal/SelectDecorator.kt
package com.example.clever.view.home.cal

import android.app.Activity
import android.graphics.drawable.Drawable
import com.example.clever.R
import com.prolificinteractive.materialcalendarview.CalendarDay
import com.prolificinteractive.materialcalendarview.DayViewDecorator
import com.prolificinteractive.materialcalendarview.DayViewFacade

class SelectDecorator(context: Activity?) : DayViewDecorator {

    private val drawable: Drawable?

    override fun shouldDecorate(day: CalendarDay?): Boolean {
        return true
    }

    override fun decorate(view: DayViewFacade?) {
        if (drawable != null) {
            view?.setSelectionDrawable(drawable)
        }
    }

    init {
        drawable = context!!.getResources().getDrawable(R.drawable.calendar_select)
    }
}